docs/DeveloperGuide.md

```diff
@@ -250,26 +250,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+`* * *` | user | edit a task| edit the task by accessing the task and typing the changes
+`* * *` | user | find a task by keywords | access details of tasks quickly without having to go through the entire list
+`* * *` | user |only view uncompleted task in CMDo| avoid being confused by completed task that are overdue| 
+`* * *` | user | my tasks sorted by due date and due time|locate urgent tasks easily | 
+`* * *` | user | remove my completed tasks| see only uncompleted tasks | 
+`* * *` | user | book by time slots| block out time for unconfirmed events |
+`* * *` | user | undo my previous action | make mistakes | 
+`* * *` | user | redo my previous action upon undo | make mistakes |
+`* * *` | user | set priority to my task | know which task is more important | 
+`* *` | user |Simply assign a date due to my todo by typing in the due date| to enter due dates easily
+`* *` | user |block out time slots of unconfirmed tasks| avoid scheduling tasks that clash
+`* *` | user |auto reschedule a task i am unable to complete due at the moment| save the effort of manual rescheduling
 
 ## Appendix B : Use Cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is `CMDo` and the **Actor** is the `user`, unless specified otherwise)
+
+### Use case: Add a task
+
+**MSS**
+
+1. User requests to add a task
+2. CMDo stores the natural language into data
+3. CMD0 adds the task <br>
+>Use case ends.
+
+**Extensions**
+
+1a. The given input is invalid
+
+> 1a1. CMDo shows help message <br>
+  Use case resumes at step 1
 
-#### Use case: Delete person
+2a. Date is not specified
+
+> CMDo stores the task as a floating one
+  Use case resumes at step 3	
+
+2b. Date is specified but time is not
+
+> CMDo stores the task with the input date and the time would be 2359 for that date.
+  Use case resumes at step 3
+
+2c. Priority is not specified
+
+> CMDo stores it as low priority
+  Use case resumes at step 3
+
+2d. Two priorities are specified
+
+> CMDo stores it as the highest priority indicated (eg. low, high, high is stored as the priority)
+   Use case resumes at step 3
+  
+### Use case: Block out time slot
 
 **MSS**
+  
+  
+1. User requests to block time slot
+2. CMDo stores the natural language into data
+3. CMD0 blocks the specified time slot <br>
+>Use case ends.
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
+**Extensions**
+
+1a. The given input is invalid
+
+> 1a. CMDo shows help message <br>
+  Use case resumes at step 1
+
+2a. Date is not specified
+
+> 2a. CMDo shows help message <br>
+  Use case resumes at step 1
+
+2b. Date is specified but time is not
+
+> 2b. CMDo shows help message <br>
+  Use case resumes at step 1
+
+2c. Time slot clash
```

Comment:
hmmmm wouldn't it be more intuitive to show all the "free" timeslots instead of all the blocked? of course show block times along with it would be fine too ~ but rmb to always be able to justify ur decisions ~